packages/api/lib/granule-delete.js

```diff
@@ -27,6 +44,7 @@ const deleteGranuleAndFiles = async ({
 }) => {
   if (pgGranule === undefined) {
     // Delete only the Dynamo Granule and S3 Files
+    await _deleteS3Files(dynamoGranule.files);
     await granuleModelClient.delete(dynamoGranule);
   } else {
```

Comment:
The logic here is the reverse of what's below. For Dynamo granules we:

1. Delete the S3 files
2. Delete the Dynamo granule record

For PG granules we delete the S3 files last.

Open to switching this but it's this way because the original Dynamo Granule delete had it structured in this order. The idea was just to keep it as close to as-is as possible. Functionally though, I don't see a reason it couldn't be reversed if anyone feels strongly.